### PR TITLE
Fix: All Windows Pop-up scaling on hidpi monitors

### LINKED ISSUE #44: Issue with "All Windows" and Pop over applications randomly starts having wrong offsets
This issue occurs after a number of the same type of windows (8+) are opened, which is pretty common for me with Terminal windows (lots of remote ssh connections).  

By default if you right-mouse click on a icon in the dock, one of the options is "All Windows".      
image
The bottom is random where it is rendered, sometimes it is rendered higher and sometimes lower.   The more windows you open  the higher it will render upwards and so you get a nice "empty" space between the dock and the menu...   (This is a full height image from top of screen to the dock.  So the cut-off on top is where it is cut-off on the screen.

---

Related is that if you enable: "minimize-or-previews" on the click action, it also mis-renders and looks like this: (full width of screen image):
image -- it always renders it off the left side.    The wider the title the less room left-right distance is and the quicker this rendering issue occurs on the previews, so it can be duplicated easier.

It appears that the logic for rendering doesn't have any cap for max height ( or left) so that it goes negative coords, which then the more you add, the more negative the rendering goes.

### PR DESCRIPTION
**Pull request by 13r0ck**
rebase of #48 onto jammy
closes #44 

@NathanaelA I made minor changes to your PR, does this still look good to you?

### CODE REVIEW

**Comment by NathanaelA** [comment]
Looks good to me.   I'll test it shortly.

File: windowPreview.js [hunk 1]
```diff
@@ -36,10 +36,12 @@ var WindowPreviewMenu = class DashToDock_WindowPreviewMenu extends PopupMenu.Pop
         this._source = source;
         this._app = this._source.app;
         let monitorIndex = this._source.monitorIndex;
-
-        this.actor.add_style_class_name('app-menu');
-        this.actor.set_style('max-width: '  + (Main.layoutManager.monitors[monitorIndex].width  - 22) + 'px; ' +
-                             'max-height: ' + (Main.layoutManager.monitors[monitorIndex].height - 22) + 'px;');
+        let scale = St.ThemeContext.get_for_stage(global.stage).scale_factor;
+    
+        this.actor.add_style_class_name('app-well-menu');
```
Comment (NathanaelA) [inline windowPreview.js]: Not that I think this is something wrong,  what is the difference between the original `app-menu` on line 40 with the new `app-well-menu`.   What difference does this create?

**Comment by NathanaelA** [comment]
Works great for me in Hi-Def mode.   Trying to switch to Low-Def mode on my screen confused my system and so I wasn't able to test it (probably need to reboot).   But in Hi-Def both items are perfectly aligned and don't wrap off the screen.

File: windowPreview.js (hunk 1 above)
Comment (13r0ck) [reply windowPreview.js]: I'm not actually sure, but for some reason that was changed up stream. Good catch! I went ahead and updated your commit with that change

**Review by n3m0-22 — CHANGES_REQUESTED** [review]
Tested on multiple HiDPI and LoDPI monitors. As mentioned before by @jacobgkau in [#48](https://github.com/pop-os/cosmic-dock/pull/48) it is not perfectly centered, but this does not impact functionality. On both Hi and LoDPI screens at a variety of resolutions it seems to have resolved previous issues. I was not able to test it with fraction scaling. The one issue I found, and this occurred on both HiDPI and LoDPI screens can be reproduced as follows.

1. Unplug and replug the monitor or reset GNOME shell
2. Right click on the icon in the dock

The first time doing this the All Windows section will extend off the screen and has no scroll bar. Try the same thing again and now the box is contained on the screen and has a scroll bar.

Examples on HiDPI and LoDPI:
HiDPI_01
HiDPI_02
LoDPI_01
LoDPI_02

**Comment by NathanaelA** [comment]
Do we know why this hasn't been merged yet @n3m0-22 -- This is extreamly annoying that each time I get a new version of Cosmic Dock this is totally borked again and I have to manually fix it...

**Comment by leviport** [comment]
@NathanaelA The reason for not merging it was described. That's why changes were requested.

**Comment by n3m0-22** [comment]
@NathanaelA  The reason it has not been merged is due to the issue I listed above.

**Comment by NathanaelA** [comment]
Does this PR I wrote actually create a new issue, or is this something that you found that is already present in the code base because of the sizing is also based on the size of the screen?.  The only thing this PR changes is that if the theme gives us a scaling it will apply it.    So I took this to mean that you found another issue, but it wasn't actually caused by this PR.     

**However, if you can tell me for sure it is the new scaling code is the issue I can give you a quick fix.   As I know exactly what could have created it in this PR.**  :grinning: 

I unfortunately do NOT have an external monitor, and I just tested using "ALT-F2, R" to restart the shell multiple times and could not duplicate this issue at all.   Every time the shell started up, both scrolling lists were correctly sized to my screen...

**Comment by n3m0-22** [comment]
As standard practice I would not mention the issue unless it is a regression caused by the PR itself, but since this is fairly old I'll be happy to double check and get back to you to be sure.

**Comment by NathanaelA** [comment]
I just created a new PR with the fix for this issue, _if_ it is the new code causing it.   I also realized why the left/right window list window wasn't centered while I was adding this new code and added the fix in the new #171 PR.    Please let me know if you have any other issues, and I'll try and get them fixed ASAP.  :grinning:

**Comment by n3m0-22** [comment]
I'll check it out. Thanks.

**Comment by n3m0-22** [comment]
Closed by: https://github.com/pop-os/cosmic-dock/pull/171

**Pull request closed without merging**